### DISCUSSION TITLE
Sharing state across different feature files

### DISCUSSION THREAD

**Discussion by AkaFroe** [badeball/cypress-cucumber-preprocessor#1078] [Q&A] [answered]
Hey all, 
I just started using the cucumber-preprocessor with cypress and running into a dead end on the following problem:

For my use case I want to create a new customer and save the customer number in a variable. In a different test I want to search for the newly created customer.

Resulting in something like this:

```
Then("I create a new customer", function () {
  // some code to create the customer
    cy.get("table").contains("th", "Client number:").next().then(($clientnumber) => {
        let number = $clientnumber.text()
       cy.wrap(number).as('clientNumber')
    })
})
```

In a different step that is used in a different test I want to use the above number to search for the customer or when that variable is empty use a hardcoded number

```
Then("I can search for a customer", function () {
    cy.get(input[class='customer-number']).type(clientNumber ? clientNumber : "202196")
})
```

This works when I run both scenario's in the same feature file but I would like to split it up and save the extracted number for later use for other scenario's

I looked at the documentation but unfortunately I cant make much sense of it 
https://github.com/badeball/cypress-cucumber-preprocessor/blob/master/docs/state-management.md

**Comment by badeball** [comment, marked as answer]
The context is thrown away between feature files / specs and this is the case preprocessor or not. You can eg. use tasks to persist / maintain state in a plugin. However, sharing context between feature files sounds like a horrendously bad idea (well-established anti-pattern).

**Reply by AkaFroe** [reply]
Ah didnt know that. I probably mixed some information up then. 
Thanks for the quick answer, Will make it one feature file then with multiple scenario's
